### DISCUSSION TITLE
Install latest version on Raspberry Pi

### DISCUSSION THREAD

**Discussion by mizamae** [django/channels#1700] [Help!]
HEllo,

I am trying to install the latest channels version (3.0.5) into a Raspberry and I found the problem of the dependency to the cryptography >3.4 package. This package requires a rust toolchain to generate the wheel installable and the RaspPi does not support it.

My question is, what is  the latest version of Channels  that requires cryptography version below 3??

P.D.: this incompatibility with RaspPi is the only drawback I can now find in channels latest versions

**Comment by smithdc1** [comment]
I've managed to install rustup / rust on a raspberry pi 400 to build cryptography. Takes a _while_ to build, but it works.

**Comment by yedpodtrzitko** [comment]
channels dont have `cryptography` as its requirement, so I'm not sure if this is the right place to ask. Anyway, you can specify lower version of `cryptography` which does not require Rust yet (ie. `3.3.2`). Works for me with the latest channels version.

**Reply by mizamae** [reply]
When I try to install channels it installs cryptography >=3.4.6... and it requires the rust thing ..

**Reply by yedpodtrzitko** [reply]
- that does not mean it's a direct requirements of `channels`. `channels` have its requirements, which have their requirements etc., so it's somewhere in the chain, not in `channels` themselves
- wherever the `cryptography` requirement comes from, it tries to install any version of `cryptography`, so it goes for the latest one. But it is not mandatory. I'm using latest version of `channels` and it installs successfully with a manually specified requirement `cryptography==3.3.2` in my `pyproject.toml` (or `requirements.txt` or whatever you use for requirements) without trying to install the latest version.
